cooltools/cli/logbin_expected.py

```diff
@@ -0,0 +1,180 @@
+         "raw Hi-C reads",
+    type=int,
+    nargs=1,
+    default=50,
+    show_default=True
+)
+@click.option(
+    "--spread-funcs",
+    metavar="spread_funcs",
+    help="A way to estimate the spread of the P(s) curves between regions. "
+    "* 'minmax' - the minimum/maximum of by-region P(s)\n"
+    "* 'std' - weighted standard deviation of P(s) curves (may produce negative results)\n "
+    "* 'logstd' (recommended) weighted standard deviation in logspace",
+    type=click.Choice(["minmax", "std", "logstd"]),
+    default='logstd',
+    show_default=True,
+    nargs=1,
+)
+@click.option(
+    "--spread-funcs-slope",
+    metavar="spread_funcs_slope",
+    help="Same as spread-funcs, but for slope (derivative) ratehr than P(s)",
+    type=click.Choice(["minmax", "std", "logstd"]),
+    default='std',
+    show_default=True,
+    nargs=1,
+)
+@click.option(
+    "--resolution",
+    metavar="resolution",
+    help="Data resolution in bp. If provided, additonal column of separation in bp "
+    "(s_bp) will be added to the outputs",
+    type=int,
+    nargs=1,
+)
+def logbin_expected(
+    expected_path,
+    output_prefix,
+    bins_per_order_magnitude,
+    bin_layout,
+    min_nvalid,
+    min_count,
+    spread_funcs,
+    spread_funcs_slope,
+    resolution
+):
+    """
+    Logarithmically bin expected values generated using compute_expected for cis data.
+    
+    This smoothes the data, resulting in clearer plots and more robust analysis results.
+    Also calculates derivative after gaussian smoothing.
+    For a very detailed escription, see
+    https://github.com/open2c/cooltools/blob/51b95c3bed8d00a5f1f91370fc5192d9a7face7c/cooltools/expected.py#L988
+
+    EXPECTED_PATH : The paths to a .tsv file with output of compute_expected.
+    Must include a header. Use the '::' syntax to specify a summary column name.
+    
+    OUTPUT_PREFIX: Output file name prefix to store the logbinned expected
+    (prefix.log.tsv) and derivative (prefix.der.tsv) in the tsv format."
+    """
+
+    # unpack expected path and name as generated by click's callback to validate_csv:
+    expected_path, exp_summary_name = expected_path
+    # that's what we expect as column names:
+    expected_columns = ["region", "diag", "n_valid", exp_summary_name]
+    # expected dtype as a rudimentary form of validation:
+    expected_dtype = {
+        "region": np.str,
+        "diag": np.int64,
+        "n_valid": np.int64,
+        exp_summary_name: np.float64,
+    }
+
+    # use 'usecols' as a rudimentary form of validation,
+    # and dtype. Keep 'comment' and 'verbose' - explicit,
+    # as we may use them later:
+    cvd = pd.read_csv(
+        expected_path,
+        usecols=expected_columns,
+        dtype=expected_dtype,
+        comment=None,
+        sep='\t',
+        verbose=False,
+    )
+
+    lb_cvd, lb_slopes, lb_distbins = expected.logbin_expected(
+        cvd,
+        exp_summary_name=exp_summary_name,
```

Comment:
makes sure other names of diagonal summaries are allowed ...
one example could be "count.sum" to calculate Pc from raw but binned data (could be useful) i guess ...